torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2379,6 +2379,25 @@ def sample_inputs_jiterator(op, device, dtype, requires_grad, **kwargs):
 
         yield SampleInput(lhs, args=tuple(args), kwargs=sample_kwargs, broadcasts_input=broadcasts_input)
 
+def sample_inputs_broadcast_shapes(op, device, dtype, requires_grad, **kwargs):
+    shapes = (
+        ((), ()),
+        ((S,), ()),
+        ((S, 1), (S,)),
+        ((M, S), ()),
+        ((S, M, S), (M, S)),
+        ((S, M, S), (S, M, S)),
+        ((M, 1, S), (M, S)),
+        ((M, 1, S), (1, M, S)),
+        ((0, 1, 3), (0, 10, 3))
+    )
+
+    # restricting test runs on cpu/bool to prevent repetitive runs
```

Comment:
Eh, I wouldn't bother for simplicity

Reply:
I'll update this along with the dtype suggestion as well.

Maybe I'm missing another attribute. I would still need to it for the dtype, right? Otherwise, `test_ops.py` would complain about

```
test_dtypes_broadcast_shapes_cpu (__main__.TestCommonCPU) ... FAIL

======================================================================
FAIL: test_dtypes_broadcast_shapes_cpu (__main__.TestCommonCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/pytorch/pytorch/torch/testing/_internal/common_device_type.py", line 377, in instantiated_test
    result = test(self, **param_kwargs)
  File "/opt/pytorch/pytorch/torch/testing/_internal/common_device_type.py", line 786, in test_wrapper
    return test(*args, **kwargs)
  File "/opt/pytorch/pytorch/torch/testing/_internal/common_device_type.py", line 821, in dep_fn
    return fn(slf, *args, **kwargs)
  File "/opt/pytorch/pytorch/torch/testing/_internal/common_device_type.py", line 979, in only_fn
    return fn(self, *args, **kwargs)
  File "test_ops.py", line 302, in test_dtypes
    self.fail(msg)
AssertionError: The supported dtypes for broadcast_shapes on device type cpu are incorrect!
The following dtypes worked in forward but are not listed by the OpInfo: {torch.bfloat16, torch.complex128, torch.float16, torch.uint8, torch.bool, torch.int8, torch.float64, torch.int16, torch.int32, torch.complex64, torch.int64}.
```

Reply:
Just xfail test_dtypes for this op with a note about why (because we're hacking the system)